silk/datadog_checks/silk/check.py

```diff
@@ -0,0 +1,224 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from copy import deepcopy
+
+from six.moves.urllib.parse import urljoin, urlparse
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.time import get_timestamp
+
+from .events import SilkEvent
+from .metrics import BLOCKSIZE_METRICS, METRICS, READ_WRITE_METRICS
+
+EVENT_PATH = "events?timestamp__gte={}"
+
+STATE_ENDPOINT = 'system/state'
+SERVERS_ENDPOINT = 'system/servers'
+
+STATE_MAP = {'online': AgentCheck.OK, 'offline': AgentCheck.WARNING, 'degraded': AgentCheck.CRITICAL}
+
+
+class SilkCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'silk'
+
+    STATE_SERVICE_CHECK = "system.state"
+    SERVERS_SERVICE_CHECK = "server.state"
+    CONNECT_SERVICE_CHECK = "can_connect"
+    METRICS_TO_COLLECT = METRICS
+
+    def __init__(self, name, init_config, instances):
+        super(SilkCheck, self).__init__(name, init_config, instances)
+
+        server = self.instance.get("host_address")
+        self.latest_event_query = int(get_timestamp())
+        self.url = "{}/api/v2/".format(server)
+
+        host = urlparse(server).netloc
+        self._tags = self.instance.get("tags", []) + ["silk_host:{}".format(host)]
+
+        if self.instance.get("enable_read_write_statistics"):
```

Comment:
Since `enable_read_write_statistics` and `enable_blocksize_statistics` are optional, should a default `False` value be added here?

Reply:
I can add a default `False` for when doing `instance.get()`, although I think runtime config would set it to `False` by default?